internal/lock/sync.go

```diff
@@ -0,0 +1,92 @@
+package lock
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"go.jetpack.io/devbox/internal/cuecfg"
+)
+
+func SyncLockfiles() error {
+	latestPackages, err := latestPackages()
+	if err != nil {
+		return err
+	}
+
+	// Step 2: Update the devbox.lock files
+	return filepath.Walk(".", func(path string, fi os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !fi.IsDir() && filepath.Base(path) == "devbox.lock" {
+			var lockFile File
+			if err := cuecfg.ParseFile(path, &lockFile); err != nil {
+				return err
+			}
+
+			changed := false
+			for key, latestPkg := range latestPackages {
+				if pkg, exists := lockFile.Packages[key]; exists {
+					if pkg.LastModified != latestPkg.LastModified {
+						lockFile.Packages[key].AllowInsecure = latestPkg.AllowInsecure
+						lockFile.Packages[key].LastModified = latestPkg.LastModified
+						// PluginVersion is intentionally omitted
+						lockFile.Packages[key].Resolved = latestPkg.Resolved
+						lockFile.Packages[key].Source = latestPkg.Source
+						lockFile.Packages[key].Version = latestPkg.Version
+						lockFile.Packages[key].Systems = latestPkg.Systems
+						changed = true
+					}
+				}
+			}
+
+			if changed {
+				if err = cuecfg.WriteFile(path, lockFile); err != nil {
+					return err
+				}
+				fmt.Printf("Updated: %s\n", path)
+			}
+		}
+		return nil
+	})
+}
+
+func latestPackages() (map[string]*Package, error) {
+	latestPackages := make(map[string]*Package)
+
+	err := filepath.Walk(".", func(path string, fi os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !fi.IsDir() && filepath.Base(path) == "devbox.lock" {
+			var lockFile File
+			if err := cuecfg.ParseFile(path, &lockFile); err != nil {
+				return err
+			}
+
+			for key, pkg := range lockFile.Packages {
+				if latestPkg, exists := latestPackages[key]; exists {
+					currentTime, err := time.Parse(time.RFC3339, pkg.LastModified)
+					if err != nil {
+						return err
+					}
+					latestTime, err := time.Parse(time.RFC3339, latestPkg.LastModified)
+					if err != nil {
+						return err
```

Comment:
Does this error if one of the lockfile is super old and does not have `LastModified` in there? (or is that scenario non existent)

Reply:
I'm not sure that ever happens, but we should handle that. I think a missing last modified should be considered old and replaced if possible.